pubsub/gossipsub/gossipsub-v1.2.md

```diff
@@ -0,0 +1,102 @@
+# Gossipsub v1.2
+
+# Overview
+
+This document aims to provide a minimal extension to the [gossipsub
+v1.1](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md)
+protocol.
+
+The proposed extensions are backwards-compatible and aim to add utility control messages for 
+better network efficiency (delivery latency and total throughput) tuning of gossip protocol 
+implementations 
+
+In more specific terms, new control messages are introduced: `PING` and `PONG`. 
+They are primarily intended for collecting statistics to estimate peer connection 
+latency and bandwidth and adjusting message broadcasting strategy accordingly.
+
+# Specification
+
+## Protocol Id
+
+Nodes that support this Gossipsub extension should additionally advertise the
+version number `1.2.0`. Gossipsub nodes can advertise their own protocol-id
+prefix, by default this is `meshsub` giving the default protocol id:
+- `/meshsub/1.2.0`
+
+## Parameters
+
+This section lists the configuration parameters that needs to agreed on across clients to avoid
+peer penalizations
+
+| Parameter         | Description                                                            | Reasonable Default |
+|-------------------|------------------------------------------------------------------------|--------------------|
+| `max_ping_rate`   | maximum number of `PING` messages per minute      | 60                 |
+
+## PING/PONG Messages
+
+### Sender
+
+Sender MAY send `PING` message (either standalone or piggy tailing) at any time. Sender is free choose 
+any `id` however it might be reasonable to select unique numbers across the session. 
+
+Sender MUST NOT send `PING` message to a peer if `max_ping_rate` `PING` message were already sent 
+to that peer during the last minute.
+
+An implementation is free to choose a reasonable timeout (e.g. 60 seconds) for waiting for a corresponding 
+`PONG` message. However it is worth considering that a `PONG` message might be 
+[... blocked by another message which is already on the wire] and not setting the timeout too small.
+If a `PONG` timeout occurs that SHOULD be treated as protocol malfunction and the peer connection 
```

Comment:
```suggestion
`PONG` timeout exceeding SHOULD be treated as protocol malfunction and the peer connection 
```

Reply:
A question: if any other message is received from a remote peer, should it reset the timeout? For instance, if a channel was busy by transmitting a big message from a slow peer a `PONG` might be too late and cause disconnection by timeout. I think defining a recommended timeout value in the spec should be useful